### PR TITLE
Improve release checklist

### PR DESCRIPTION
**Pull request by straight-shoota**
Restructures and updated the release checklist.
I think it's much clearer to read with separate sections.

### CODE REVIEW

File: processes/crystal-release.md [hunk 1]
```diff
@@ -1,43 +1,76 @@
...
+
+1. Finalize the release PR
+   * Make sure all changes are mentioned in the changelog
+   * Check release date
+   * Un-draft the PR
+2. Split off release branch (`release/x.y`)
+3. Verify Maintenance CI workflow succeeds on the HEAD of the release branch
+4. Smoke test with [test-ecosystem](https://github.com/crystal-lang/test-ecosystem)
+   * Run [*Test Crystal & Shards Workflow](https://github.com/crystal-lang/test-ecosystem/actions/workflows/test-crystal-shards.yml) with the release branch as `crystal_branch`.
+4. Merge the release PR
```
Comment (beta-ziliani) [inline processes/crystal-release.md:26]: This is one off, isn't it?

File: processes/crystal-release.md [hunk 2]
```diff
@@ -1,43 +1,76 @@
...
+### Binary releases
+
+1. Wait for the release build in circle CI
+2. Smoke test with test-ecosystem (again)
+3. Attach build artifacts to Github release
+   * `crystal-*-darwin-*.tar.gz`
+   * `crystal-*-linux-*.tar.gz`
+   * `crystal-*.pkg`
+   * `crystal-*-docs.tar.gz`
+3. Push changes to OBS for building linux packages
```
Comment (beta-ziliani) [inline processes/crystal-release.md:49]: Same

**Review by beta-ziliani — APPROVED** [review]
Just a few off-indexes, but besides that it's a great improvement

File: processes/crystal-release.md (hunk 1 above)
Comment (straight-shoota) [reply processes/crystal-release.md:26]: Did you notice all the indices of nested lists are the same? 🤫 😆

**Review by beta-ziliani — APPROVED** [review]
🚀

**Pull request merged**